File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        longest = 1 if len(nums) >= 1 else 0
        for n in nums:
            cur = 1
            if n+1 in nums and n-1 not in nums:
                n2 = n
                while n2+1 in nums:
                    cur += 1
                    longest = max(longest, cur)
                    n2 += 1
        return longest

File: Data Structures & Algorithms/integer-break/submission-2.rs
impl Solution {
    pub fn integer_break(n: i32) -> i32 {
        let mut dp = HashMap::from([(1,1), (2, 1)]);

        for num in 2..=n {
            dp.insert(num, if num == n {0} else {num});
            for k in 1..num {
                let cur = *dp.get_mut(&num).unwrap();
                let l = *dp.get(&k).unwrap();
                let r = *dp.get(&(num-k)).unwrap();
                *dp.get_mut(&num).unwrap() = max(cur, l * r)
            }
        }

        *dp.get(&n).unwrap()
    }
}
